Rerun failed updates and add to sidebar (#8173)
circleci/circleci-docs: jekyll/_cci2/rerun-failed-tests.adoc

```

The snippet above will write the list of test file names to `files.txt`.  On a non-rerun, this list will be all of the test file names.  On a "rerun", the list will be a subset of file names (the test file names that had at least 1 test failure in the previous run).  You can pass the list of test file names from `files.txt` into, for example, your custom `makefile`.

[#known-limitations]
== Known limitations

* If your testing job uses parallelism and test splitting, the job will spin up the number of containers/virtual machines (VMs) that are specified with the `parallelism` key. However, the step that runs tests for each of those parallel containers/VMs will only run a subset of tests, or no tests, after the tests are split across the total number of parallel containers/VMs.
+
For example, if parallelism is set to eight, there may only be enough tests after the test splitting occurs to "fill" the first parallel container/VM. The remaining seven containers/VMs will still start up, but they will not run any tests when they get to the test execution step.
+

> **The vast majority of customers still observe substantial time and credit savings** despite spinning up containers/VMs that do not run tests.

Reviewer: ```suggestion
**In most cases, you can still observe substantial time and credit savings** despite spinning up containers/VMs that do not run tests.
```

---

> `circleci tests run` expects input to be space or newline delimited.  If your test file names have spaces in them, this may pose a problem, especially if you are using `pytest` which may generate names with whitespace.  One possible workaround is to use specific IDs for the tests with whitespace in their names using the instructions link:https://docs.pytest.org/en/7.1.x/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test[here].

Reviewer: ```suggestion
`circleci tests run` expects input to be space or newline delimited.  If your test file names have spaces in them, this may pose a problem, especially if you are using `pytest` which may generate names with whitespace.  One possible workaround is to use specific IDs for the tests with whitespace in their names using the instructions from the link:https://docs.pytest.org/en/7.1.x/example/parametrize.html#set-marks-or-test-id-for-individual-parametrized-test[official Pytest documentation].
```

> If your job runs tests in parallel and persists files to a workspace, you may see a parallel run on a rerun that fails because the `persist_to_workspace` step could not find any contents in the directory that was specified. This may happen because the parallel run will not always execute tests on a rerun if there are not enough tests to be distributed across all parallel runs.

Reviewer: I wonder if it makes sense to link to this section from the Known Limitations point above re: parallelism? e.g. "see the troubleshooting section for avoiding parallel run failure" or something to that effect

Author: @kbatuigas I'm not sure what you're suggesting here, probably my Friday brain. Let's discuss at standup.

Author: I think I've done this now
